File: Week_02/kotlin/src/_102_BinaryTreeLevelOrderTraversal.kt
import java.util.*

class _102_Bfs {
    fun levelOrder(root: TreeNode?): List<List<Int>> {
        val ret = mutableListOf<List<Int>>()
        val queue = LinkedList<TreeNode>()
        root?.let { queue.offer(it) }
        while (queue.isNotEmpty()) {
            var size = queue.size
            val list = mutableListOf<Int>()
            while (size-- > 0) {
                val node = queue.poll()
                list += node.`val`
                node.left?.let { queue.offer(it) }
                node.right?.let { queue.offer(it) }
            }
            ret += list
        }
        return ret
    }
}

File: Week_02/cpp/src/_102_BinaryTreeLevelOrderTraversal.cpp
#include <vector>
#include <queue>
using namespace std;
struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};
class Bfs
{
public:
    vector<vector<int>> levelOrder(TreeNode *root)
    {
        vector<vector<int>> ret;
        queue<TreeNode *> q;
        if (root)
            q.push(root);
        while (!q.empty())
        {
            int size = q.size();
            vector<int> row;
            while (size-- > 0)
            {
                auto n = q.front();
                q.pop();
                row.push_back(n->val);
                if (n->left)
                    q.push(n->left);
                if (n->right)
                    q.push(n->right);
            }
            ret.push_back(row);
        }
        return ret;
    }
};